RiotSwiftUI/Modules/Room/PollHistory/View/PollHistory.swift

```diff
@@ -50,35 +46,91 @@ struct PollHistory: View {
         }
     }
     
+    @ViewBuilder
+    private var content: some View {
+        if viewModel.viewState.polls == nil {
+            loadingView
+        } else if viewModel.viewState.polls?.isEmpty == true {
+            noPollsView
+        } else {
+            pollListView
+        }
+    }
+    
     private var pollListView: some View {
         ScrollView {
             LazyVStack(spacing: 32) {
-                let enumeratedPolls = Array(viewModel.viewState.polls.enumerated())
-                
-                ForEach(enumeratedPolls, id: \.offset) { _, pollData in
+                ForEach(viewModel.viewState.polls ?? []) { pollData in
                     PollListItem(pollData: pollData)
                 }
                 .frame(maxWidth: .infinity, alignment: .leading)
-                
-                Button {
-                    #warning("handle action")
-                } label: {
-                    Text("Load more polls")
-                }
-                .frame(maxWidth: .infinity, alignment: .leading)
+
+                loadMoreButton
+                    .frame(maxWidth: .infinity, alignment: .leading)
             }
             .padding(.top, 32)
             .padding(.horizontal, 16)
         }
     }
     
+    private var loadMoreButton: some View {
+        HStack(spacing: 8) {
+            if viewModel.viewState.isLoading {
+                spinner
+            }
+            
+            Button {
+                #warning("handle action in next ticket")
+            } label: {
+                Text("Load more polls")
+            }
+            .disabled(viewModel.viewState.isLoading)
+        }
+    }
+    
+    @ViewBuilder
+    private var spinner: some View {
+        ProgressView()
+            .progressViewStyle(CircularProgressViewStyle())
+    }
+    
+    @ViewBuilder
     private var noPollsView: some View {
-        Text(viewModel.mode == .active ? VectorL10n.pollHistoryNoActivePollText : VectorL10n.pollHistoryNoPastPollText)
-            .font(theme.fonts.body)
-            .foregroundColor(theme.colors.secondaryContent)
+        if viewModel.viewState.canLoadMoreContent {
+            let days = PollHistoryConstants.chunkSizeInDays
+            
+            VStack(spacing: 32) {
+                Text(viewModel.mode == .active ? VectorL10n.pollHistoryNoActivePollPeriodText("\(days)") : VectorL10n.pollHistoryNoPastPollPeriodText("\(days)"))
+                    .font(theme.fonts.body)
+                    .foregroundColor(theme.colors.secondaryContent)
+                    .multilineTextAlignment(.center)
+                    .padding(.horizontal, 16)
+                    .accessibilityIdentifier("PollHistory.emptyLoadMoreText")
+    
+                loadMoreButton
+            }
```

Comment:
This `noPollsView` might be a bit neater with the following changes:
- Always have the `VStack` present.
- Add a computed property to the viewModel that provides the text string based upon both `mode` and `canLoadMoreContent`
- Conditionally show the `loadMoreButton` based on `viewModel.viewState.canLoadMoreContent`

Reply:
Great suggestion actually.
It looks much better now! :-)
https://github.com/vector-im/element-ios/pull/7293/commits/3e9251433a5aa18326d518cd114866691f76bfba